### PR TITLE
FX converter documentation

### PR DESCRIPTION
**Pull request by apbose**

### CODE REVIEW

File: docsrc/contributors/fx_converters.rst [hunk 1]
```diff
@@ -0,0 +1,133 @@
+.. _conversion:
```
Comment (narendasan) [inline docsrc/contributors/fx_converters.rst:1]: this label is going to conflict with other files, use something more specific

File: docsrc/contributors/fx_converters.rst [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+==================
+The converter library in Torch-TensorRT is located in ``TensorRT/py/torch_tensorrt/fx/converters``.
+They are categorized into - ``aten_ops_converters``, ``acc_ops_converters`` and ``nn_ops_converters``.
+The individual converters present are useful for the quantization workflow.
+The converters are registered using the ``tensorrt_converter``
+
+Steps
+==================
+
+External Interface
```
Comment (narendasan) [inline docsrc/contributors/fx_converters.rst:13]: Rename this `Op Sets`, it doesnt really matter how they are generated (though it is useful to mention this) but the high order bit is that there are different op sets that we could register converters for.

File: docsrc/contributors/fx_converters.rst [hunk 3]
```diff
@@ -0,0 +1,133 @@
+.. _conversion:
+
+FX Converters
+==================
+The converter library in Torch-TensorRT is located in ``TensorRT/py/torch_tensorrt/fx/converters``.
+They are categorized into - ``aten_ops_converters``, ``acc_ops_converters`` and ``nn_ops_converters``.
+The individual converters present are useful for the quantization workflow.
+The converters are registered using the ``tensorrt_converter``
```
Comment (narendasan) [inline docsrc/contributors/fx_converters.rst:8]: Mention what the impl library is

**Review by narendasan — COMMENTED** [review]
@apbose Generally this does not feel like it is explaining the methodology for creating a converter, it sort of references a few implementations we already have. But the point of this doc is to explain to a user outside the library how they might write a converter. 

So for example explaining the `@torch_tensorrt.fx.tensorrt_converter` and the expected function signature like you do is a good start but lean into the idea of being a user outside the code base trying to write a converter. 

Also explain the `impl` library more in this framing as well, i.e. a library of existing functional blocks that can be composed to create a converter.

File: docsrc/contributors/fx_converters.rst [hunk 4]
```diff
@@ -0,0 +1,195 @@
+.. _conversion:
+
+FX Converters
```
Comment (narendasan) [inline docsrc/contributors/fx_converters.rst:3]: Lets refocus this to Dynamo Converters

File: docsrc/contributors/fx_converters.rst [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+==================
+
+Operation Sets
+-------------------
+There are three different converter sets for FX in torch_tensorrt. Depending on whether the operation is generated using acc_trace, aten_trace or fx_trace, the converters are categorized to one of the three operation sets - 
+``aten_ops_converters``, ``acc_ops_converters`` or ``nn_ops_converters``.  The converters are registered using ``tensorrt_converter`` decorator. The function decorated
+has the arguments - ``network, target, args, kwargs, name``,  which is common across all the operators schema.
+These functions are mapped in the ``aten`` converter registry dictionary (at present a compilation of FX and dynamo converters, FX will be deprecated soon), with key as the function target name.
+    
+    * acc_ops_converters
```
Comment (narendasan) [inline docsrc/contributors/fx_converters.rst:20]: Since we are focusing on dynamo, we only need to talk about aten converters

File: docsrc/contributors/fx_converters.rst [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+        *  acc_trace is produced by ``torch_tensorrt.fx.tracer.acc_tracer.acc_tracer.trace``.
+    * aten_ops
+        There are two options at present for this
+            #.   Dynamo: aten_trace is produced by ``torch_tensorrt.dynamo.backend.compile``. The second round of trace is produced by ``aot_torch_tensorrt_aten_backend`` by invoking ``aot_module_simplified`` from ``torch._functorch.aot_autograd``
+            #.   FX: aten_trace is produced by ``torch_tensorrt.fx.tracer.dispatch_tracer.aten_tracer.trace``. This flow is more common currently, but this will soon be deprecated in torch_tensorrt.
```
Comment (narendasan) [inline docsrc/contributors/fx_converters.rst:25]: Talk to @peri044 about the new workflow for dynamo.compile

File: docsrc/contributors/fx_converters.rst [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+    * nn_ops
+        *  symbolic_trace is produced by ``torch.fx._symbolic_trace``.
+
+The implementation of the above converter set is to be included in the common implementation library present in ``TensorRT/py/torch_tensorrt/fx/impl``.
```
Comment (narendasan) [inline docsrc/contributors/fx_converters.rst:29]: dynamo/conversion/converters/impl

File: docsrc/contributors/fx_converters.rst [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+This documentation focuses on the implementation of the ``aten_ops`` converters. There might be some steps involved in reorganizing files for ``acc_ops`` converters. This is discussed in more detail in the next section.
+
+Converter implementation
+------------------------
+In this section, we illustrate the steps to be implemented for writing a converter. We divide them according to activation, operator or lowering pass implementation.
+Each of them is detailed with the help of an example
+
+    * Registration
+        
+        The converter needs to be registered with the appropriate op code in the ``tensorrt_converter``.
```
Comment (narendasan) [inline docsrc/contributors/fx_converters.rst:39]: dynamo_tensorrt_converter

File: docsrc/contributors/fx_converters.rst [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+
+            Example: ``leaky_relu``
+            
+            * acc_ops_converters
+                
+                Define in ``py/torch_tensorrt/fx/converters/acc_ops_converters``. One needs to register the opcode generated in the trace, with ``tensorrt_converter`` decorator. Op code to be used for the registration or the converter registry key in this case is ``acc_ops.leaky_relu``
+
+                .. code-block:: python
+
+                    @tensorrt_converter(acc_ops.leaky_relu)
```
Comment (narendasan) [inline docsrc/contributors/fx_converters.rst:51]: Not needed any more

File: docsrc/contributors/fx_converters.rst [hunk 10]
```diff
@@ -0,0 +1,195 @@
...
+
+    * Implementation Library
+
+        The converters across all the above three opsets have the common implementation library ``py/torch_tensorrt/fx/converters/impl``
+    
+        * Activation
+
+            Example: ``leaky_relu``
+        
+            The implementation is to be placed in present in ``py/torch_tensorrt/fx/impl/activation.py``. This is where all the activation functions are defined and implemented.
```
Comment (narendasan) [inline docsrc/contributors/fx_converters.rst:110]: dynamo impl lib

File: docsrc/contributors/fx_converters.rst [hunk 11]
```diff
@@ -0,0 +1,195 @@
...
+                return torch.add(
+                    torch.mul(input_, beta), torch.mul(torch.matmul(mat1, mat2), alpha)
+                )
+
+
+       
+Tests
+-----
+
+* FX testing:
```
Comment (narendasan) [inline docsrc/contributors/fx_converters.rst:176]: We dont need info on fx

File: docsrc/contributors/fx_converters.rst (hunk 6 above)
Comment (apbose) [reply docsrc/contributors/fx_converters.rst:28]: @peri044 could you mention some points here?

File: docsrc/contributors/fx_converters.rst [hunk 12]
```diff
@@ -0,0 +1,211 @@
...
+The dynamo converter library in Torch-TensorRT is located in ``TensorRT/py/torch_tensorrt/dynamo/conversion``.
+
+
+
+Steps
+==================
+
+Operation Set
+-------------------
+The converters in dynamo are produced by ``aten_trace`` and falls under ``aten_ops_converters`` ( FX earlier had ``acc_ops_converters``, ``aten_ops_converters`` or ``nn_ops_converters`` depending on the trace through which it was produced).  The converters are registered using  ``dynamo_tensorrt_converter`` for dynamo. The function decorated
```
Comment (gs-olive) [inline docsrc/contributors/fx_converters.rst:14]: Since only `torch.export` uses `aten_trace`, consider rewording as:

> Our Dynamo paths use a set of PyTorch operators called ATen, which are produced by tracers which translate operators into this core set.

File: docsrc/contributors/fx_converters.rst [hunk 13]
```diff
@@ -0,0 +1,211 @@
...
+has the arguments - ``network, target, args, kwargs, name``,  which is common across all the operators schema.
+These functions are mapped in the ``aten`` converter registry dictionary (at present a compilation of FX and dynamo converters, FX will be deprecated soon), with key as the function target name.
+    
+    * aten_trace is produced by ``torch_tensorrt.dynamo.backend.compile``. The second round of trace is produced by ``aot_torch_tensorrt_aten_backend`` by invoking ``aot_module_simplified`` from ``torch._functorch.aot_autograd``
```
Comment (gs-olive) [inline docsrc/contributors/fx_converters.rst:18]: Could be reworded to imply that `export` and `compile` trace slightly differently but both result in the same key set of ops.

> The traced ATen graph is produced by either `torch_tensorrt.dynamo.trace` (in the export path), or `aot_module_simplified` (in the compile path). Both of these tracers reduce the large set of Torch operators to the more concise ATen set.

File: docsrc/contributors/fx_converters.rst [hunk 14]
```diff
@@ -0,0 +1,211 @@
...
+            Example: ``leaky_relu``
+            
+            
+            * aten_ops_converters: Dynamo_converters
+
+                    Define in ``py/torch_tensorrt/dynamo/conversion/aten_ops_converters``. One needs to register the opcode generated in the trace with ``dynamo_tensorrt_converter`` decorator. Op code to be used for the registration or the converter registry key in this case is ``torch.ops.aten.leaky_relu.default``
+
+                        .. code-block:: python
+            
+                            @dynamo_tensorrt_converter(torch.ops.aten.leaky_relu.default)
```
Comment (gs-olive) [inline docsrc/contributors/fx_converters.rst:45]: Add a tab/indent here, or unindent the following code

File: docsrc/contributors/fx_converters.rst [hunk 15]
```diff
@@ -0,0 +1,211 @@
...
+
+
+
+       
+Tests
+-----
+
+* Dynamo testing: 
+    
+    Dynamo tests are present for the lowering ops in ``py/torch_tensorrt/dynamo/backend/test/test_decompositions.py``. The above converters will soon be ported to dynamo tests
```
Comment (gs-olive) [inline docsrc/contributors/fx_converters.rst:202]: This has been moved to `tests/py/dynamo/lowering/test_decompositions.py`

**Review by gs-olive — COMMENTED** [review]
The additional documentation is great. Added some comments on wording, code, and file location updates.

File: docsrc/contributors/fx_converters.rst [hunk 16]
```diff
@@ -0,0 +1,211 @@
...
+Steps
+==================
+
+Operation Set
+-------------------
+The converters in dynamo are produced by ``aten_trace`` and falls under ``aten_ops_converters`` ( FX earlier had ``acc_ops_converters``, ``aten_ops_converters`` or ``nn_ops_converters`` depending on the trace through which it was produced).  The converters are registered using  ``dynamo_tensorrt_converter`` for dynamo. The function decorated
+has the arguments - ``network, target, args, kwargs, name``,  which is common across all the operators schema.
+These functions are mapped in the ``aten`` converter registry dictionary (at present a compilation of FX and dynamo converters, FX will be deprecated soon), with key as the function target name.
+    
+    * aten_trace is produced by ``torch_tensorrt.dynamo.backend.compile`` or  ``torch_tensorrt.dynamo.backend.export``.
```
Comment (narendasan) [inline docsrc/contributors/fx_converters.rst:18]: This isnt the case anymore right? its torch_tensorrt.dynamo.compile and torch.compile(backend="torch_tensorrt")

Comment (apbose) [reply docsrc/contributors/fx_converters.rst]: @narendasan @gs-olive . Changed this. Could you verify once?

**Review by narendasan — APPROVED** [review]
LGTM

**Review by gs-olive — APPROVED** [review]
Looks good to me!
